### PR TITLE
Add missing CHIP error codes into CHIPError.cpp/TestCHIPErrorStr.cpp and fix duplicates

### LINKED ISSUE #17191: Some chip errors defined in CHIPError.h have not been added to CHIPError.cpp and TestCHIPErrorStr.cpp
#### Problem
Some chip errors defined in `CHIPError.h` have not been added to `CHIPError.cpp` and `TestCHIPErrorStr.cpp`

For example: `CHIP_ERROR_INVALID_FILE_IDENTIFIER`

 #### Proposed Solution

Need to find all chip errors defined in `CHIPError.h` that are missing this change and update the translation of error codes to strings in `CHIPError.cpp` and add them to `kTestElements[]` in `TestCHIPErrorStr.cpp`

### PR DESCRIPTION
**Pull request by isiu-apple**
#### Problem

- Some CHIP errors defined in `CHIPError.h` have not been added to `CHIPError.cpp` and `TestCHIPErrorStr.cpp`
- Some CHIP errors have conflicting error codes

Fixes: https://github.com/project-chip/connectedhomeip/issues/17191

#### Change overview
- Added missing CHIP errors to `CHIPError.cpp` and `TestCHIPErrorStr.cpp`
- Resolve conflicting CHIP error definitions

#### Testing

Verified Linux OTA-R compilation.
Verified `src/lib/core/tests/BUILD.gn` compilation

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by carol-apple — APPROVED** [review]

File: src/lib/core/CHIPError.cpp [hunk 1]
```diff
@@ -620,12 +626,21 @@ bool FormatCHIPError(char * buf, uint16_t bufSize, CHIP_ERROR err)
...
+        desc = "Fabric mismatch on ICA";
+        break;
     case CHIP_ERROR_MESSAGE_COUNTER_OUT_OF_WINDOW.AsInteger():
         desc = "Message id out of window";
         break;
     case CHIP_ERROR_REBOOT_SIGNAL_RECEIVED.AsInteger():
         desc = "Termination signal is received";
         break;
+    case CHIP_ERROR_NO_SHARED_TRUSTED_ROOT.AsInteger():
+        desc = "No shated trusted root";
```
Comment (msandstedt) [inline src/lib/core/CHIPError.cpp:642]:
```suggestion
        desc = "No shared trusted root";
```

File: src/lib/core/CHIPError.cpp [hunk 2]
```diff
@@ -407,6 +404,15 @@ bool FormatCHIPError(char * buf, uint16_t bufSize, CHIP_ERROR err)
     case CHIP_ERROR_INVALID_FABRIC_ID.AsInteger():
         desc = "Invalid Fabric Id";
         break;
+    case CHIP_ERROR_TOO_MANY_CONNECTIONS.AsInteger():
+        desc = "Too many connections";
+        break;
+    case CHIP_ERROR_SHUT_DOWN.AsInteger():
+        desc = "The operation cancelled because a shut down was initiated";
```
Comment (msandstedt) [inline src/lib/core/CHIPError.cpp:411]:
```suggestion
        desc = "The operation was cancelled because a shut down was initiated";
```

**Review by harsha-rajendran — APPROVED** [review]

File: src/lib/core/CHIPError.cpp (hunk 2 above)
Comment (isiu-apple) [reply src/lib/core/CHIPError.cpp]: Fixed.

File: src/lib/core/CHIPError.cpp (hunk 1 above)
Comment (isiu-apple) [reply src/lib/core/CHIPError.cpp]: Fixed.

**Pull request merged**